ColorSwitcher/Extensions/GIFViewController+GIFManagerDelegate.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  GIFViewController+GIFManagerDelegate.swift
+//  ColorSwitcher
+//
+//  Created by Matt M Smith on 10/5/20.
+//
+
+import SwiftyGif
+import UIKit
+
+extension GIFViewController: GIFManagerDelegate {
+    func getGIFModel(title: String, urlString: String) {
+        guard let url = URL(string: urlString) else {
+            let errorMessage = "Sorry, but we are having issues getting your GIF"
+            let alert = UIAlertController(title: "ERROR", message: errorMessage, preferredStyle: .alert)
+            let action = UIAlertAction(title: "OK", style: .default)
+            alert.addAction(action)
+            
+            present(alert, animated: true)
+            
+            return
+        }
+        
+        DispatchQueue.main.async { [weak self] in
```

Comment:
Small suggestion: There's nothing wrong at all about what you coded here. You'll see the same in some Kroger code. However, we landed on a standard of grabbing a reference to self inside closures where they are needed like this:

`guard let strongSelf = self else { return }`

...then using `strongSelf` in place of `self?`. There are a couple of reasons. First, the `self` really isn't Optional at the point of use, and it isn't really `self` either, it's a reference to `self`. The `guard` acts as a barrier and also a visual cue that this isn't really going to work without a reference to `self`. 

If you review our MRs and don't see it, call us on it. :)

Reply:
Good call dispatching back to main.